### DISCUSSION TITLE
Disallow downloading and reading files of hosting via Github.

### DISCUSSION THREAD

**Discussion by ZEkA10000** [community/community#48474] [New to GitHub]
Yesterday I wondered, it's possible to make github to disable read files and download repository if this repository uses as free hosting. 
I just worry if someone can download my "website" and clone it.

**Comment by valentinojosh** [comment]
Hi!
If you are using GitHub to host your site then it is not possible to 100% disable read access or prevent someone from from downloading/cloning the repo. 

There is still some things you can do to aid this though. 

One option is to set the repository to private. Sometimes this is not possible depending on how you are hosting it, however, if possible it is a great way to prevent unauthorized users from directly accessing your repository.

Besides that you can really only attempt to hide sensitive information. I recently got an email from a tool called, GitGuardian, notifying me I had a secrets leak in one of me repositories. It's a helpful tool for that type of scenario.

Thats really the two best options. Hope that helps!

**Reply by queenofcorgis** [reply]
@valentinojosh thanks for this great answer and taking the time to respond. @ZEkA10000 if this helped you, don't forget to mark as answer for anyone else with this question ✅
